pkg/v1/providers/infrastructure-vsphere/v1.3.1/ytt/overlay.yaml

```diff
@@ -69,6 +38,41 @@
 #@   return 0
 #@ end
 
+
+#@ valid_pci_devices = [[0x10DE,0x1EB8]]
+
+#@ def valid_pci_devices_list():
+#@       return ",".join(map(lambda x:  "0x%X:0x%X" % (x[0], x[1]), valid_pci_devices))
+#@ end
+
+#@ def get_pci_devices(pci_devices_string, pci_ignore_pci_device_validation):
+#@     pci_devices = list()
+#@     help_error = "VSPHERE_CONTROL_PLANE_PCI_DEVICES \"" + pci_devices_string + "\" must be in the  \"0x<vendorId>:0x<deviceId>,0x<vendorId>:0x<deviceId>,..\" format."
+#@     for val in pci_devices_string.split(","):
+#@         if len(val) <= 0:
+#@             assert.fail(help_error)
+#@         end
+#@         kv = val.split(":")
+#@         if len(kv) != 2:
+#@             assert.fail(help_error)
+#@         end
+#@         id_pair = dict()
+#@         vendor_id = int(kv[0], 16)
+#@         device_id = int(kv[1], 16)
```

Comment:
Should we return specific error message if the deviceId or vendorId provided is not a valid hexadecimal?

Reply:
There's no try-catch support for this in Starlark unfortunately.

Reply:
Added a comment that this needs to be covered in documentation.